transport_nantes/topicblog/views.py

```diff
@@ -654,3 +654,64 @@ class TopicBlogPressSend(LoginRequiredMixin, TemplateView):
 
     """
     pass
+
+
+######################################################################
+# TopicBlogLauncher
+
+class TopicBlogLauncherViewOnePermissions(PermissionRequiredMixin):
+    """Custom Permission class to require different permissions
+    depending on whether the user is requesting a GET or a POST.
+
+    Default behaviour is at class level and doesn't allow a
+    per-method precision.
+    """
+    def has_permission(self) -> bool:
+        user = self.request.user
+        if self.request.method == 'POST':
+            return user.has_perm('topicblog.tbla.may_publish')
+        elif self.request.method == 'GET':
+            return user.has_perm('topicblog.tbla.may_view')
+        return super().has_permission()
+
+class TopicBlogLauncherEdit(TopicBlogBaseEdit):
+    model = TopicBlogLauncher
+    template_name = 'topicblog/tb_launcher_edit.html'
+    form_class = TopicBlogItemForm
+
+
+class TopicBlogLauncherView(TopicBlogBaseView):
+    model = TopicBlogLauncher
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context['context_appropriate_base_template'] = 'topicblog/base_launcher.html'
+        tb_object = context['page']
+        user = self.request.user
+        if user.has_perm('topicblog.tbp.may_send_self') or \
+           (user.has_perm('topicblog.tbp.may_send') \
+            and tb_object.publisher != user):
+            context['sendable'] = True
+        return context
+
+
+class TopicBlogLauncherViewOne(TopicBlogLauncherViewOnePermissions,
+                            TopicBlogBaseViewOne):
+    model = TopicBlogLauncher
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        return context
+
+
+class TopicBlogLauncherList(TopicBlogBaseList):
```

Comment:
No permissions set here, like Press, TBOList doesn't set permissions, we need to add the appropriate permissions here